### PR TITLE
DID Upgrade Documentation

### PR DESCRIPTION
**Pull request by geel9**
Adds documentation and sample code for DID Upgrades

### CODE REVIEW

File: docs/reference/services/provider-service.md [hunk 1]
```diff
@@ -270,6 +270,70 @@ Deletes a webhook from an ecosystem.
...
+    Many DID methods are based on distributed ledgers (commonly known as "blockchains"), so the following should be kept in mind when registering a DID which uses a ledger:
+
+    - Depending on the DID method chosen, registering a DID may incur a ledger write fee, which is different for each ledger. **You are responsible for these fees.**
+        - We will provide clear documentation regarding these fees before this feature becomes generally available.
+    - Newly-registered DIDs may not be immediately resolvable, as most ledgers cannot provide instantaneous writes.
+        - Issuance and verification of credentials against these DIDs will still function during this time, but only within the Trinsic platform.
+
+Upgrades a wallet's DID from the default `did:key` to another DID Method. This endpoint may only be called by an ecosystem provider.
+
+Trinsic will register a DID Document for you, and update the wallet's `public_did` property to the newly-registered DID.
```
Comment (janpieterz) [inline docs/reference/services/provider-service.md]: Could you maybe write a sentence or two about:
1. It not being possible to reverse this, once done it's out there
1. The impact (e.g. what about already issued credentials from this wallet)
1. The benefits?

**Review by janpieterz — APPROVED** [review]
lgtm, little nitpicks re adding some context to the docs but just throw something on there that feels good to you ;)

File: dotnet/Tests/Tests.cs [hunk 2]
```diff
@@ -26,12 +26,12 @@ namespace Tests;
...
 public class Tests
 {
-#if DEBUG
+#if DEBUG && false
     private const string DefaultEndpoint = "localhost";
     private const int DefaultPort = 5000;
     private const bool DefaultUseTls = false;
 #else
-    private const string DefaultEndpoint = "staging-internal.trinsic.cloud";
+    private const string DefaultEndpoint = "dev-internal.trinsic.cloud";
```
Comment (tmarkovski) [inline dotnet/Tests/Tests.cs]: We test SDK against staging. Forgot to revert this or is there another reason?

File: dotnet/Tests/Tests.cs [hunk 3]
```diff
@@ -26,12 +26,12 @@ namespace Tests;
 [SuppressMessage("ReSharper", "MethodHasAsyncOverload")]
 public class Tests
 {
-#if DEBUG
+#if DEBUG && false
```
Comment (tmarkovski) [inline dotnet/Tests/Tests.cs]: Intentional disable?

Comment (fundthmcalculus) [inline dotnet/Tests/Tests.cs]: This prevents localhost ever being used, is that intentional? Also, default should be staging, not dev for testing purposes.

File: dotnet/Tests/Tests.cs (hunk 2 above)
Comment (fundthmcalculus) [inline dotnet/Tests/Tests.cs]:
```suggestion
    private const string DefaultEndpoint = "staging-internal.trinsic.cloud";
```

File: dotnet/Tests/Tests.cs [hunk 4]
```diff
@@ -26,12 +26,12 @@ namespace Tests;
...
     private const string DefaultEndpoint = "localhost";
     private const int DefaultPort = 5000;
     private const bool DefaultUseTls = false;
```
Comment (fundthmcalculus) [inline dotnet/Tests/Tests.cs]:
```suggestion
    private const string DefaultEndpoint = "dev-internal.trinsic.cloud";
    private const int DefaultPort = 443;
    private const bool DefaultUseTls = true;
```

File: dotnet/Tests/Tests.cs (hunk 3 above)
Comment (fundthmcalculus) [inline dotnet/Tests/Tests.cs]:
```suggestion
#if DEBUG
```

File: dotnet/Tests/Tests.cs (hunk 2 above)
Comment (fundthmcalculus) [reply dotnet/Tests/Tests.cs]: Beat me to it! 😁

Comment (geel9) [reply dotnet/Tests/Tests.cs]: Accidental inclusion -- thanks for catching this

**Pull request merged**